tools/tests/test_osqueryd.py

```diff
@@ -66,5 +67,28 @@ def test_3_daemon_with_watchdog(self):
         # dies when the watcher goes away
         self.assertTrue(daemon.isDead(children[0]))
 
+    def test_4_daemon_sighup(self):
+        # A hangup signal should not do anything to the daemon.
+        daemon = self._run_daemon({
+            "disable_watchdog": True,
+        })
+        self.assertTrue(daemon.isAlive())
+
+        # Send a SIGHUP
+        os.kill(daemon.proc.pid, signal.SIGHUP)
+        self.assertTrue(daemon.isAlive())
+
+    def test_4_daemon_sigint(self):
+        # An interrupt signal will cause the daemon to stop.
+        daemon = self._run_daemon({
+            "disable_watchdog": True,
```

Comment:
Yes! All of the children (extensions/worker) will receive the `SIGINT` and turn down their event handling. The watcher will begin to wait for a child to interrupt and return a `-2`, not the expected `-14`. A third test should be added to make sure the daemon does in fact abort after safely waiting for the worker. 

This third test could be implemented using a check for `-2` OR the signal handling logic could be modified to allow the watcher to remember the handled signal and return `-14` after the safe wait.